### PR TITLE
Fix a bug in slice global inits where guard expressions were disregarded

### PR DESCRIPTION
**Pull request by danpoe**

### CODE REVIEW

File: src/goto-programs/slice_global_inits.cpp [hunk 1]
```diff
@@ -77,6 +77,8 @@ void slice_global_inits(goto_modelt &goto_model)
     {
       const codet &code=i_it->code;
       code.visit(visitor);
+      const exprt &expr=i_it->guard;
+      expr.visit(visitor);
```
Comment (tautschnig) [inline src/goto-programs/slice_global_inits.cpp]: While at it, may I challenge the existence of `symbol_collectort`? I'd recommend using `find_symbols` instead.

Comment (danpoe) [reply src/goto-programs/slice_global_inits.cpp]: Ok, I'll use `find_symbols`. Out of curiosity: What are the `ID_next_symbol`'s that it also collects (when called with `next=true`)?

Comment (tautschnig) [reply src/goto-programs/slice_global_inits.cpp]: `ID_next_symbol` is used by tools like SatAbs that talk about a current vs. a next state (on paper these are typically the primed variables).

**Comment by danpoe** [comment]
@tautschnig  I think this can be approved if the fixes look ok now.

**Review by tautschnig — APPROVED** [review]

**Pull request merged**